### ISSUE TITLE
Multiple Reductions broken by erroneous GROUPBY insertion

### ISSUE THREAD

**Issue by slorello89** [redis/redis-om-dotnet#64]
When trying to run multiple reductions during an aggregation, e.g.:

```csharp
return _aggregations.GroupBy(x => x.RecordShell.State)
            .Sum(x => x.RecordShell.Salary)
            .Average(x => x.RecordShell.Age).ToList();
```

The ExpressionTranslator is erroneously pushing a 0 argument `GROUPBY` onto the predicate stack before the second reduction to guard against a reduction without a `GroupBy`, this is causing the second reduction to fail as the information has not been loaded into the pipeline.

Why is this happening? Prior to adding a reduction to the pipeline, the ExpressionTransalator checks if a GroupBy is already on the predicate stack, this is because running a reduction without a groupby will result in a failure. As of right now [this operation](https://github.com/redis/redis-om-dotnet/blob/main/src/Redis.OM/Common/ExpressionTranslator.cs#L371) is only peeking back 1 predicate and only checking if that predicate is a `GROUPBY`, it should be sufficient to still peek back 1, but check if the previous Aggregation Predicate is a GROUPBY OR a Reduction.

**Comment by slorello89** [comment]
Should be fixed in 0.1.6

**Comment by lzegiestowskiTAG** [comment]
Great, thanks for the help.
